.github/workflows/test.yml

```diff
@@ -0,0 +1,47 @@
+name: KBase File Cache Server test
+
+on:
+  pull_request:
+    types:
+    - opened
+    - reopened
+    - synchronize
+    - ready_for_review
+  push:
+    # run workflow when merging to main or develop
+    branches:
+      - main
+      - master
+      - develop
+
+jobs:
+  file_cache_server_tests:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Repo checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.9.19
```

Comment:
Why such a low version?

Reply:
Most of the other repos where we have done the Mongo upgrade are using this version. I'm trying to make it consistent here. If you prefer 3.11.x, that's fine. I don't have a strong opinion on this.

Reply:
I don't remember off the top of my head, but I think there were specific reasons why we used lower versions. If it's just as easy to go to a higher version go ahead and do it. Also make sure the test version and build version are aligned

Reply:
If we change the build version to 3.9.19, 19/39 tests would fail. If we want to git them fixed, I prefer to do them in the next PR. If we upgrade python version to 3.11.x, expect more dep changes because of build error.

Reply:
This still needs to be updated in the build I assume

Reply:
yes, so we need to create an issue for that as well.

Reply:
no, it needs to be updated in the build in this PR - the build and tests need to be aligned

Reply:
E.g. https://github.com/kbase/file_cache_server/blob/develop/Dockerfile#L1 and maybe other placees

Reply:
I commented out the mypy check to allow the tests to run. Let me know if you know how to fix [945b204](https://github.com/kbase/file_cache_server/pull/24/commits/945b204efefcd84e1d0c8de9e08256a2dc514b32)

Reply:
In my experience mypy isn't worth the trouble as long as you have reasonable tests. I like what it's trying to do but it seems too clunky (although I haven't tried it recently, maybe it's better now)